Project/searchResult.js

```diff
@@ -0,0 +1,310 @@
+        searchValue.id = "searchValue"
+        searchValue.placeholder = "Type search here"
+        searchValue.style.width = "100%"
+        inputAndSearchContainer.classList.add("d-flex")
+        inputAndSearchContainer.style.paddingBottom = ("1rem")
+        searchButton.type = "submit"
+        searchButton.id = "button"
+        searchButton.innerText = "Search"
+        searchButton.classList.add("btn", "btn-info", "btn-sm", "text-center")
+        spinner.id = "loadingSpinner"
+        spinner.classList.add("spinner-border", "d-none")
+
+        // Construct the properties of the class
+        this.listContainer = document.getElementById('listContainer')
+        this.searchButton = document.getElementById("button");
+        this.searchValue = document.getElementById("searchValue")
+        this.spinner = document.getElementById('loadingSpinner')
+        this.compareButton;
+
+        // Append list container to its parent
+        this.element.append(listContainer)
+        searchContainer.append(this.element)
+
+        // Event listener that sends a fetch based on input value (including the autosearch debounce feature)
+        this.searchValue.addEventListener('keyup', this.debounce(this.getStockData.bind(this), 1000))
+
+
+    }
+
+
+
+    // Fetch that returns response from API
+    async getStockData() {
+        let searchInput = searchValue.value
+
+        // Get the URL
+        let url = window.location.href;
+
+        // This is the starting URL
+        let baseUrl = window.location.origin + window.location.pathname;
+
+        // Reset the URL after each "Search"
+        history.replaceState(null, '', baseUrl);
+
+        // Change URL to add query string in milestone 2.2 but use an IF statement to make sure if the input is blank to return nothing and stop the function
+        let newUrl = baseUrl + `?query=${searchInput}`;
+        if (searchInput === "") {
+            history.pushState(null, '', baseUrl);
+            return;
+        } else {
+            history.pushState(null, '', newUrl);
+        }
+
+        this.enableSpinner()
+
+        const response = await fetch(`https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/search?query=${searchInput}&limit=10&exchange=NASDAQ`)
+        const data = await response.json()
+
+        // Fetch the full company detail by looping through each symbol from the search result
+        async function fetchFullCompanyDetail(key) {
+            const response2 = await fetch(`https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/company/profile/${key}`)
+            const data2 = await response2.json()
+            return data2
+        }
+
+        // Create an array to put in results of the full company detail
+        let results = [];
+
+        const createArrayWithFullData = async () => {
+            for (const key of data) {
+                const result = await fetchFullCompanyDetail(key.symbol)
```

Comment:
instead of wating for a request to finish in order to start new one, use promise.all() to send all the requests in the same time.